### DISCUSSION TITLE
No release notes? What is "MultiHeadAttention to return qk as well"

### DISCUSSION THREAD

**Discussion by turnkit** [openai/whisper#822] [Q&A] [answered]
This update was pushed last week as well as another but there are no announcements, no release notes.

Would someone explain what these changes do?

"MultiHeadAttention to return qk as well"

- https://github.com/openai/whisper/commit/53807677fe8bc892fe0fff970e8b24ddd3848596

Also:
"word-level timestamps in Multilingual_ASR notebook"

- https://github.com/openai/whisper/commit/28769fcfe50755a817ab922a7bc83483159600a9

**Comment by glangford** [comment, marked as answer]
It looks like the "MultiHeadAttention to return qk as well" change enables the demo code in the Multilingual_ASR notebook. This is mentioned (briefly) in https://github.com/openai/whisper/discussions/332.

A comment in the notebook describes the purpose as

> ...we use the cross-attention weights to determine more granular, word-level timestamps. It uses a set of heuristics and dynamic time warping (DTW) to find the alignment between the audio and the transcript.

The accuracy of timestamps has been discussed in https://github.com/openai/whisper/discussions/3 and https://github.com/openai/whisper/discussions/435 .

**Reply by glangford** [reply]
The inspiration for the notebook probably goes back to this comment from a few months ago
https://github.com/openai/whisper/discussions/3#discussioncomment-3703465
